app/jobs/sync_mailing_lists_job.rb

```diff
@@ -33,6 +32,10 @@ def perform
     treasurers = UserGroup.officers.flat_map(&:active_roles).filter { |role| role.metadata.status == RolesMetadataOfficers.statuses[:treasurer] }
     GsuiteMailingLists.sync_group("treasurer@worldcubeassociation.org", treasurers.map(&:user).map(&:email))
 
+    UserGroup.board.each do |board|
```

Comment:
You configured `GroupsMetadataBoard` under the assumption that there is only one Board, but then here you suddenly pretend like we can have 15 boards with their own separate email syncs.
Please try to focus on one approach only and be consistent with it